### ISSUE TITLE
[📚] Which is right? - analytics().logScreenView({})

### ISSUE THREAD

**Issue by suzu-prog** [invertase/react-native-firebase#4214]
## Documentation Feedback

I don't know which one is the right answer...

Is `screen_class` first?
Or is `screen_name` first?

And logScreenView({}) works...?
There may be something wrong with the way it's working.

https://github.com/invertase/react-native-firebase/blob/81c4e3b7d2ce956ed6a2cc2f40a323ec5379d6a1/packages/analytics/lib/index.d.ts#L800-L807

https://github.com/invertase/react-native-firebase/blob/81c4e3b7d2ce956ed6a2cc2f40a323ec5379d6a1/packages/analytics/lib/index.js#L110-L113

https://github.com/invertase/react-native-firebase/blob/81c4e3b7d2ce956ed6a2cc2f40a323ec5379d6a1/packages/analytics/e2e/analytics.e2e.js#L123
---

**Comment by mikehardy** [comment]
It is an object, they are named properties. It does not matter which one you set first, so long as you set them by name:

```javascript
  analyticsScreen(screenName: string): void {
    firebase.analytics().logScreenView({ screen_name: screenName, screen_class: screenName });
  }
```

In that method I explicitly set both to the same thing, but you see how with the object properties named it would not matter either way?

**Comment by suzu-prog** [comment]
@mikehardy 
I'm sorry.
```
  analytics().logScreenView({
    screen_class: 'hogeScreen',
    screen_name: 'hogeScreen',
  });
```
I wrote this, but it doesn't show up in the DebugView in the Firebase Console.


```
  analytics().logEvent({
    'hogeEvent',
    {id: 'event_id'
 });
````
The logEvent() was displayed in the DebugView.

**Comment by mikehardy** [comment]
You provide no details on what platform you are having trouble with, so I'm just guessing on what docs to show
However, how did it go when you turned on debugging on the device? 

https://firebase.google.com/docs/analytics/get-started?platform=ios#start_logging_events

```
To view this event in the Xcode debug console, enable Analytics debugging:

    In Xcode, select Product > Scheme > Edit scheme...
    Select Run from the left menu.
    Select the Arguments tab.
    In the Arguments Passed On Launch section, add -FIRAnalyticsDebugEnabled.
```

Other than that, I'm not sure. If the issue persists you can follow two strategies (in parallel perhaps):

1. reach in to node_modules directy and either set breakpoints or add print-debugging to the native method calls to verify they are doing what they should be doing:

 https://github.com/invertase/react-native-firebase/blob/55cd752f6c1c1ab063e1cb9235d3657c5b2ae27f/packages/analytics/lib/index.js#L219-L222

https://github.com/invertase/react-native-firebase/blob/55cd752f6c1c1ab063e1cb9235d3657c5b2ae27f/packages/analytics/ios/RNFBAnalytics/RNFBAnalyticsModule.m#L38-L53

2. Make a pure native example against the upstream APIs in order to provide a reproduction to the upstream team (e.g. https://github.com/firebase/firebase-ios-sdk) - https://github.com/firebase/quickstart-ios/tree/master/analytics

Then report your findings, if there is something to do in this module we can get it fixed

**Comment by serhangursoy** [comment]
I didn't want to create a new issue for this because it feels like I'm missing something. Perhaps you can help @mikehardy 

I'm having a problem with the example project after the update. 
Currently I'm getting the following error
```log
 ERROR    TypeError: (0, _analytics.default)().logScreenView is not a function. (In '(0, _analytics.default)().logScreenView({
          screen_name: screenName,
          screen_class: screenName
        })', '(0, _analytics.default)().logScreenView' is undefined)
```
Code is simply this
```js
import analytics from '@react-native-firebase/analytics';

class AnalyticsService{
    
    static SendTelemetry(key, data){
        analytics().logEvent(key,data);
    }

    static SetCurrentScreen(screenName, override){
        analytics().logScreenView({
            screen_name: screenName,
            screen_class: screenName
        });
    }

}

export default AnalyticsService;
```

`SendTelemetry` method works without any issue however problem is arising at `SetCurrentScreen` method. Before the update/depreciation I was using `analytics().setCurrentScreen(screenName, override)` and it was also working fine ( actually still working fine in production )


Package Version
```json
"@react-native-firebase/analytics": "^7.6.2",
"@react-native-firebase/app": "^8.4.2",
```

**Comment by mikehardy** [comment]
@serhangursoy I suppose you have outdated native library dependencies or code
check to see if you have overrides back to incompatible versions of firebase sdks: https://rnfirebase.io/#overriding-native-sdk-versions
check to see if build cache layers are incoherent by deleting them: `npx react-native-clean-project`

may help?

**Comment by serhangursoy** [comment]
@mikehardy  Thanks for the assist but I don't have anything override that part as far as I can see, and react-native-clean-project doesn't work properly on my Windows - although I manually cleaned up `./gradlew clean` as well and deleted the `node_modules` and reinstalled again, still no luck. 

Strange part is that, when I want to check library method, Visual Studio finds it without any issue. So Typescript-wise project suppose to have that method
I even added `    implementation 'com.google.firebase:firebase-analytics:17.5.0' ` manually just to be sure that I have the correct version - still same error.

**Comment by mikehardy** [comment]
Ah - I didn't realize react-native-clean-project does not work on Windows. That's a shame, it's very useful. 
It sounds like your code on disk is correct, but there is something stale somewhere in the chain from code to running binary on device

**Comment by serhangursoy** [comment]
I tried nearly everything - still the same issue. I will probably leave it - I can live with only having event log for now. Still, thanks for trying 🙂

**Comment by cecodor** [comment]
> I tried nearly everything - still the same issue. I will probably leave it - I can live with only having event log for now. Still, thanks for trying 🙂

Did you solve it? I am having the exact same issue :(

**Comment by serhangursoy** [comment]
> > I tried nearly everything - still the same issue. I will probably leave it - I can live with only having event log for now. Still, thanks for trying 🙂
> 
> Did you solve it? I am having the exact same issue :(

Unfortunately, no. I wasn't using that metric extensively anyway so it didn't want to mess up my already working metrics for that.

**Comment by victorsouza-mnz** [comment]
I've upgraded the lib from 6.3.x to 10.0.0 and i have this problem, i alread tried npx "react-native-clean-project" with no success , and its just logScreenView that is broken

**Comment by mikehardy** [comment]
@victorsouza-mnz I hate to say it, but v10 is from the year 2020, the likelihood of it working successfully on current mobile operating system versions is slim to "nope". You should definitely upgrade to current as you can - here's how:

https://invertase.io/blog/react-native-firebase-versioning

With regard to this specific error, that logScreenView is not found on the analytics object, that was a change made in the v10  release, and if the underlying SDKs are a match for the javascript, all from from react-native-firebase/app package then it should be working

**Comment by victorsouza-mnz** [comment]
This artcle would be insanely useful yesterday hahaha
I managed to make it work in v10. 
I started working with an app that hasn't been updated much for the last few years so I'm updating the libs little by little
